components/QueueEdit.js

```diff
@@ -0,0 +1,165 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import {
+  Col,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  Form,
+  FormGroup,
+  Input,
+  Label,
+  Button,
+  FormFeedback,
+} from 'reactstrap'
+
+const fields = [
+  {
```

Comment:
We'll need to validate location for fixed-location queues.

Reply:
Done.